src/CoreWCF.NetTcp/tests/Services/TestService.cs

```diff
@@ -50,5 +54,34 @@ public TestMessage TestMessageContract(TestMessage testMessage)
                 Body = new MemoryStream(Encoding.UTF8.GetBytes(text + " from server"))
             };
         }
+
+       
+        public string EchoForPermission(string echo)
+        {
+            IPrincipal principal = Thread.CurrentPrincipal;
+            if (echo.Contains(PrincipalPermissionMode.UseWindowsGroups.ToString()))
+            {
+                if (principal.Identity !=null
+                     && principal.Identity.GetType().Equals(typeof(WindowsIdentity)))
+                    return echo;
```

Comment:
It might be better to use `typeof(WindowsIdentity).IsAssignableFrom(principal.Identity?.GetType())`. That way we don't break if NegotiateStream starts returning a type derived from WindowsIdentity and you can remove the null check.

Reply:
Why not "is WindowsIdentity"?

Reply:
That works too 😄